### PR TITLE
DnssecDnsHandle: also update the RRSIG's proof

### LINKED ISSUE #2274: test that AD bit is set when valid answer includes DNSSEC records
to be done after #2253

the query in this test

https://github.com/hickory-dns/hickory-dns/blob/04c65accda4e9e31c8fc1f1225154cd22ea973a1/conformance/packages/conformance-tests/src/resolver/dnssec/scenarios/secure.rs#L59

does not ask for DNSSEC records so the answer only contains a single A record.

we should add a similar test that sets the `DO` bit in the query (`dnssec()` method in the `DigSettings` API) to request RRSIG records in addition to the A record.

in this case the resolver should still set the AD in the answer. the logic added in #2253 may not be sufficient to get that result. tweak if necessary

### PR DESCRIPTION
**Pull request by japaric**
fixes #2274

the reason queries with DO=1 were getting an AD=0 response was that `DnssecDnsHandle` was update the `proof` fields of all records in the response except for the RRSIG records, which should inherit the result of the DNSSEC validation.

~~this PR depends on PR #2287 so leaving it in draft state for now to prevent it being merged into the wrong branch~~

### CODE REVIEW

**Review by djc — APPROVED** [review]

**Review by justahero — APPROVED** [review]
LGTM

**Comment by japaric** [comment]
@djc merge policy question: should PRs like these and #2297 and #2300 wait on two reviews from the hickory team or is it fine to merge them with one review from the hickory team and one review from the ferrous team?

**Comment by djc** [comment]
I generally think that one approval per team is sufficient. Outside the conformance tests, if I don't feel confident enough to rely on my approval by itself, I will make that explicit (as I have in the past).

**Pull request merged**
